cm_sdk/client.py

```diff
@@ -0,0 +1,60 @@
+import requests
+from cm_sdk.api import APIException
+
+class CloudManagerClient(object):
+    """
+    Simple client for interacting over HTTP with the the MongoDB
+    CloudManager API.
+    """
+
+    HEADERS = {'Content-type': 'application/json'}
+
+    def __init__(self, api_host, group_id, api_user, api_key):
+        self.base_url = api_host + "/api/public/v1.0"
+        self.group_id = group_id
+        self.http_digest_auth = requests.auth.HTTPDigestAuth(api_user, api_key)
+
+
+    def _get_final_url(self, url_template):
+        return self.base_url + url_template.format(group_id=self.group_id)
+
+
+    def _valid_response(self, response):
+        if 'error' in response:
```

Comment:
If we're going to behave differently for a 404 vs a 500 in the sdk I think it's worth checking otherwise, it seems like forwarding this onward is better and we should do that for all errors.  The error json pattern is pretty common so the current code deals with errors that we don't know about yet in a pretty reasonable way in my opinion.